microbench/src/test/java/io/netty/microbench/channel/ImmediateLocalServerChannel.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.microbench.channel;
+
+import io.netty.channel.ChannelProgressivePromise;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.local.LocalServerChannel;
+
+public class ImmediateLocalServerChannel extends LocalServerChannel {
+    @Override
+    public ChannelPromise newPromise() {
```

Comment:
@louiscryan (and others)
A few things to consider if `AbstractChannel.newPromise()` delegates to `executor`:
1. If the channel is not registered to the event loop, then you will no longer be able to generate a promise from it. -> The user could manually create a `DefaultChannelPromise` with the channel as the argument.  Still a change we are pushing on the user.
2. Signatures don't line up between [AbstractChannel.newPromise](https://github.com/netty/netty/blob/master/transport/src/main/java/io/netty/channel/AbstractChannel.java#L292) and [EventExecutor.newPromise](https://github.com/netty/netty/blob/master/common/src/main/java/io/netty/util/concurrent/EventExecutor.java#L79). Return type difference and no argument to take the channel into account. This may mean more involved interface changes to resolve this...any ideas?